File: app/src/main/java/com/irishinterest/irishinterest/network/api/irishInterest/observer/ModelToGui.java
package com.irishinterest.irishinterest.network.api.irishInterest.observer;

import java.util.Observer;

public interface ModelToGui {
  void publishMessage(GuiMessage guiMessage);
  void registerObserverModel(Module module, Observer o);
  void unregisterObserverModel(Module module, Observer o);
}

File: app/src/main/java/com/irishinterest/irishinterest/network/api/irishInterest/observer/ModelToGui.kt
package com.irishinterest.irishinterest.network.api.irishInterest.observer

import java.util.*

interface ModelToGui {
    fun publishMessage(guiMessage: GuiMessage<*>)
    fun registerObserverModel(module: Module, o: Observer)
    fun unregisterObserverModel(module: Module, o: Observer)
}